### PR TITLE
nu-cmd-extra crate infrastructure in place with the Bits command as the model for adding other commands

### PR DESCRIPTION
**Pull request by stormasm**
I wanted to get the infrastructure in place for starters for our *nu-cmd-extra* crate...

The plan is to put inside here the following commands...

* bits
* bytes
* math

I thought it would be easier to do one at a time as well as get the nu-cmd-extra crate out there on crates.io
for this upcoming release...

Once this lands the infrastructure will be in place to move over the other noted commands for now...
And then add other stuff we do NOT want to be in 1.0.

### CODE REVIEW

File: crates/nu-cmd-extra/src/example_test.rs
Comment (amtoine) [inline crates/nu-cmd-extra/src/example_test.rs:1]: so this file is required to run the examples with the little `mod tests` at the end of the `Command` implementations, right?

**Review by amtoine — COMMENTED** [review]
yeaaah, loooks great, as with the dataframes :ok_hand: 

no tests disabled and looks it's working :muscle: 
```
cargo run --features "" -- -n
```
does not have the `bits` commands but
```
cargo run --features extra -- -n
```
does :partying_face: 

:point_down: a little question in a thread down below

:question: and an extra question, why do we have the code in `nu-cmd-extra/src/extra/` and not `nu-cmd-extra/src/`? :open_mouth:

File: crates/nu-cmd-extra/src/example_test.rs
Comment (stormasm) [reply crates/nu-cmd-extra/src/example_test.rs:1]: correct you can see that if you go ahead and remove the file and try compiling just the nu-cmd-extra crate...

all the examples need *test_examples*

Comment (amtoine) [reply crates/nu-cmd-extra/src/example_test.rs:1]: perfect :+1: 

it's the first time i see it for real, not in the `mod tests`, so i was curious :wink:

**Comment by stormasm** [comment]
>> and an extra question, why do we have the code in nu-cmd-extra/src/extra/ and not nu-cmd-extra/src/

I tried to model it after the dataframes crate...

But in reality I tried doing exactly what you said and just have 

```rust
nu-cmd-extra/src/
```

But the way *features* work in Rust and Cargo it makes it way easier to have that "extra" level of indirection, no pun intended with the word extra 😄

**Comment by amtoine** [comment]
@stormasm 
mm ok i see, that's fine and we can always get rid of that "extra" level later :relieved:

**Review by amtoine — APPROVED** [review]

**Comment by stormasm** [comment]
> @stormasm mm ok i see, that's fine and we can always get rid of that "extra" level later 😌

@amtoine 

The way features work its easy to create a feature in rust if the code lives in a particular directory...

So for example later on we can come along and bucket things up differently...

We may possibly want to have a math feature and a bits-bytes feature...

But initially I think it will be easier just to lump all of the nu-cmd-extra commands into one feature...

I always find features to be confusing to end user developers who are not intimately familiar
with the code base...

So my thinking for now is lets just keep this *simple* --- and have one feature called *extra*...

In the future if we have lots and lots of commands in nu-cmd-extra then we can come back
and re-address the number of features we want to complicate the nushell community with 😄

**Comment by amtoine** [comment]
@stormasm 
that sounds perfect to me :blush:

**Comment by amtoine** [comment]
@stormasm 
can you confirm me that this is a breaking change, as it requires people in need of the `bits` commands to build with `--features extra`? :yum:

**Comment by stormasm** [comment]
Yes it is a breaking change as they will need to build with this command to get the bits command...

```rust
cargo build --features=extra'
```

**Comment by amtoine** [comment]
thanks :relieved:

**Pull request merged**